oai_agents/agents/agent_utils.py

```diff
@@ -65,32 +66,55 @@ def __init__(self, obs_space):
         self.observation_space = obs_space
 
 class CustomAgent():
```

Comment:
CustomAgent and CustomPolicy do not clearly deliver the feature of this kind of agent and policy. We need a more specific name for them. Since it works with trajectory, I think TrajectoryFollower may be a good name?

I also asked ChatGpt to brainstorm some potential names:
HeatMapAgent: Clearly indicates that the agent's behavior is influenced by a heatmap.
TrajectoryFollower: Highlights that the agent follows a defined trajectory, which could be heatmap-driven.
PathGuideAgent: Emphasizes path selection based on the heatmap guidance.
HeatTrailAgent: Suggests that the agent moves along a "trail" indicated by heatmap data.
PrefPathAgent: Abbreviates "Preferred Path Agent," indicating it moves along a preferred trajectory.

Reply:
The purpose of `CustomAgent` is to define an agent that has a custom behavior defined by us. Now, this behavior could be to follow a heatmap region or other behaviors that we might want to have in the future. Currently, heatmap logic resides in `heatmap.py`, while the `predict` function in `CustomAgent` executes the desired behavior.
Right now, two behaviors exist: staying still or moving, controlled by `self.is_dynamic `in `CustomAgent`. As more behaviors are added, predict can evolve into:
```
if agent_type == static_adv: predict_static()
elif agent_type == dynamic_adv: predict_dyn()
elif agent_type == onion picker: ...
elif agent_type == pot blocker: ...
etc ...
```
So to sum up, I chose the names CustomAgent and CustomPolicies to highlight their flexibility for defining diverse behaviors beyond just heatmap logic.